cmd/acra-server/common/listener.go

```diff
@@ -253,31 +264,47 @@ func (server *SServer) processConnection(connection net.Conn, callback *callback
 	callback.callbackFunc(ctx, clientID, wrappedConnection)
 }
 
-func (server *SServer) start(listener net.Listener, callback *callbackData, logger *log.Entry) {
+func (server *SServer) start(parentContext context.Context, listener net.Listener, callback *callbackData, logger *log.Entry, errCh chan<- error) {
 	logger.Infof("Start listening connections")
 	for {
-		connection, err := listener.Accept()
-		if err != nil {
-			if nerr, ok := err.(net.Error); ok && nerr.Timeout() {
-				logger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorConnectionDroppedByTimeout).
-					Errorln("Stop accepting new connections due net.Timeout")
-				return
-			}
-			logger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCantAcceptNewConnections).
-				Errorln("Can't accept new connection")
+		if err := parentContext.Err(); err != nil {
+			// this means that parentContext is Done.
+			// And here we are not interesting in particular reasons of it, so just return
 			return
-		}
```

Comment:
Since the positive branch of this `if` ends with a `return`, you don't really need explicit `else`. That makes the code easier to read, and it will probably result in a better diff in this PR.

Also, I'd log the error since it may be interesting why the context has been closed.

Reply:
actually to check is context done or not better to use .Done() channel:
```
select {
case <-parentContext.Done():
  return
default:
  break
}
```
if context not finished then default branch will allow to continue execution

Reply:
Actually, yes. You are both right